fastlane/Fastfile

```diff
@@ -71,28 +80,61 @@ lane :distribute_release_build do |_options|
   release_tag = get_required_env('BUILDKITE_TAG')
   builds = distribute_builds(release_tag:)
 
-  slack(
-    username: 'CI Bot',
-    icon_url: 'https://octodex.github.com/images/jenktocat.jpg',
-    message: ":tada: #{release_tag} is now available! :tada:",
-    channel: '#dotcom-studio',
-    success: true,
-    slack_url: get_required_env('SLACK_WEBHOOK'),
-    payload: {
-      ':studio-app-icon-mac: Release available for': builds.each_value.map { |b| "[#{b[:name]}](#{b[:cdn_url]})" }.join(', ')
-    },
-    default_payloads: []
+  unless DRY_RUN
+    slack(
+      username: 'CI Bot',
+      icon_url: 'https://octodex.github.com/images/jenktocat.jpg',
+      message: ":tada: #{release_tag} is now available! :tada:",
+      channel: '#dotcom-studio',
+      success: true,
+      slack_url: get_required_env('SLACK_WEBHOOK'),
+      payload: {
+        ':studio-app-icon-mac: Release available for': builds.each_value.map { |b| "[#{b[:name]}](#{b[:cdn_url]})" }.join(', ')
+      },
+      default_payloads: []
+    )
+  end
+end
+
+def upload_to_s3_with_dry_run(bucket:, key:, file:, if_exists:, auto_prefix:)
```

Comment:
Have you considered updating the `upload_to_s3` signature to include a `dry_run:` parameter?

I ask because I don't see any remaining call to the original `upload_to_s3` implementation.